packages/perseus-core/src/parse-perseus-json/perseus-parsers/widget.ts

```diff
@@ -33,13 +34,13 @@ export function parseWidget<Type extends string, Options>(
 export function parseWidgetWithVersion<
     Version extends {major: number; minor: number} | undefined,
     Type extends string,
-    Options,
+    Options extends object,
 >(
     parseVersion: Parser<Version>,
     parseType: Parser<Type>,
     parseOptions: Parser<Options>,
 ): Parser<WidgetOptions<Type, Options>> {
-    return object({
+    return objectWithAllPropertiesRequired({
```

Comment:
We have to use `objectWithAllPropertiesRequired` because the type parameters `Type` and `Options` interact poorly with `OptionalizeProperties`. Since the concrete types of `Type` and `Options` aren't known, I think TS gives up and assumes they're `any`, which means `undefined` is assignable to them. This causes them to be incorrectly treated as optional. I don't know why TS doesn't look at the `extends` constraints for these type parameters, but it seems not to.

Note that this means we will have to use the `foo: string | undefined` syntax for the properties of `WidgetOptions`. Hopefully that's not too big a deal.

Reply:
I was trying to figure this out as I read the PR.  If I make incremental types, I see that the return type seems to be missing the `"type"` key entirely. 🤔  

<img width="966" alt="image" src="https://github.com/user-attachments/assets/f1cb6ec0-fdcd-48d5-83b5-65e8f753dbb4" />